src/render-instruction.ts

```diff
@@ -157,15 +159,19 @@ ${typeMapperImports.join('\n')}`.trim()
   }
 
   private renderIxAccountKeys(processedKeys: ProcessedAccountKey[]) {
-    const requireds = processedKeys.filter((x) => !x.optional)
-    const optionals = processedKeys.filter((x, idx) => {
-      if (!x.optional) return false
-      assert(
-        idx >= requireds.length,
-        `All optional accounts need to follow required accounts, ${x.name} is not`
-      )
-      return true
-    })
+    const requireds = this.defaultOptionalsToProgramId
+      ? processedKeys
+      : processedKeys.filter((x) => !x.optional)
+    const optionals = this.defaultOptionalsToProgramId
+      ? []
```

Comment:
The way it works now, optional accounts are always at the end. Since we don't split up the optional and required, the optional should remain in the same order they come with.

Reply:
I think potentially renaming the required and optional vars to something else would be best?

Reply:
I think the confusion here is another argument to split these strategies into separate methods ...